ViewImpl.cpp

```diff
@@ -43,85 +45,15 @@ ErrorOr<NonnullOwnPtr<LadybirdViewImpl>> LadybirdViewImpl::create(LadybirdWebVie
-
-    int ui_fd_passing_fd = fd_passing_socket_fds[0];
-    int wc_fd_passing_fd = fd_passing_socket_fds[1];
-
-    if (auto child_pid = TRY(Core::System::fork()); child_pid == 0) {
-        TRY(Core::System::close(ui_fd_passing_fd));
-        TRY(Core::System::close(ui_fd));
-
-        auto takeover_string = TRY(String::formatted("WebContent:{}", wc_fd));
-        TRY(Core::System::setenv("SOCKET_TAKEOVER"sv, takeover_string, true));
-
-        auto webcontent_fd_passing_socket_string = TRY(String::number(wc_fd_passing_fd));
-
-        ErrorOr<void> result;
-        for (auto const& path : candidate_web_content_paths) {
-            constexpr auto callgrind_prefix_length = 3;
-
-            if (Core::System::access(path, X_OK).is_error())
-                continue;
-            auto arguments = Vector {
-                "valgrind"sv,
-                "--tool=callgrind"sv,
-                "--instr-atstart=no"sv,
-                path.bytes_as_string_view(),
-                "--webcontent-fd-passing-socket"sv,
-                webcontent_fd_passing_socket_string
-            };
-            if (enable_callgrind_profiling == WebView::EnableCallgrindProfiling::No)
-                arguments.remove(0, callgrind_prefix_length);
-
-            result = Core::System::exec(arguments[0], arguments.span(), Core::System::SearchInPath::Yes);
-            if (!result.is_error())
-                break;
-        }
-
-        if (result.is_error())
-            warnln("Could not launch any of {}: {}", candidate_web_content_paths, result.error());
-        VERIFY_NOT_REACHED();
-    }
-    TRY(Core::System::close(wc_fd_passing_fd));
-    TRY(Core::System::close(wc_fd));
-
-    auto socket = TRY(Core::LocalSocket::adopt_fd(ui_fd));
-    TRY(socket->set_blocking(true));
-
-    auto new_client = TRY(adopt_nonnull_ref_or_enomem(new (nothrow) WebView::WebContentClient(move(socket), view_impl)));
-    new_client->set_fd_passing_socket(TRY(Core::LocalSocket::adopt_fd(ui_fd_passing_fd)));
-
-    if (enable_callgrind_profiling == WebView::EnableCallgrindProfiling::Yes) {
-        dbgln();
-        dbgln("\033[1;45mLaunched WebContent process under callgrind!\033[0m");
-        dbgln("\033[100mRun `\033[4mcallgrind_control -i on\033[24m` to start instrumentation and `\033[4mcallgrind_control -i off\033[24m` stop it again.\033[0m");
-        dbgln();
-    }
-
-    return new_client;
-}
-
 void LadybirdViewImpl::create_client(WebView::EnableCallgrindProfiling enable_callgrind_profiling)
 {
-    (void)enable_callgrind_profiling;
-    // auto candidate_web_content_paths = get_paths_for_helper_process("WebContent"sv).release_value_but_fixme_should_propagate_errors();
-    Vector<String> candidate_web_content_paths;
-    candidate_web_content_paths.append("./WebContent"_string.release_value_but_fixme_should_propagate_errors());
+    auto candidate_web_content_paths = MUST(get_paths_for_helper_process("WebContent"sv));
```

Comment:
Not related to this change as such, but I wonder whether it will find `/app/lib/WebContent`?

Reply:
No, but when using the serenity install in the follow-up PR, it will be installed in /app/bin anyway. It is a binary after all, not a library.

Reply:
.. but also as part of this PR, I made it install WebContent to bin as well. that's what this means

```

install(TARGETS ladybird WebContent
  EXPORT ladybirdTargets
  RUNTIME
    COMPONENT ladybird_Runtime
    DESTINATION ${CMAKE_INSTALL_BINDIR}
```

Install the runtime components of ladybird and  WebContent into CMAKE_INSTALL_BINDIR

Reply:
> it will be installed in /app/bin anyway. It is a binary after all, not a library.

It is a binary, but not one that is meant to be tun by the user from the shell. `file-hierarchy(7)` says:

> `/usr/bin/`
> Binaries and executables for user commands that shall appear in the `$PATH` search path. It is recommended not to place binaries in this directory that are not useful for invocation from a shell (such as daemon binaries); these should be placed in a subdirectory of `/usr/lib/` instead.
>
> `/usr/lib/`
> Static, private vendor data that is compatible with all architectures (though not necessarily architecture-independent). Note that this includes internal executables or other binaries that are not regularly invoked from a shell. Such binaries may be for any architecture supported by the system. Do not place public libraries in this directory, use `$libdir` (see below), instead.

Reply:
okay, that's cool, but upstream ladybird drops it in bin/. Imo it would be best to just match what we do in the cmake there and patch it properly in ladybird if that's the 'better' place for it.

Reply:
Sure, let's fix it in upstream Ladybird.